src/pages/fieldManagement/settings/defaultFieldComponents/SpeciesEditor.jsx

```diff
@@ -30,12 +30,33 @@ export default function SpeciesEditor({
   const intl = useIntl();
   const [searchInput, setSearchInput] = useState('');
   const [searchTerm, setSearchTerm] = useState(null);
+  const [stillGeneratingDisplay, setStillGeneratingDisplay] =
+    useState(false);
   const {
     data: searchResults,
     loading: searchResultsLoading,
     error: searchResultsError,
+    setError,
+    setLoading,
   } = useItisSearch(searchTerm);
 
+  const timerRef = useRef(null);
+  useEffect(() => {
+    if (searchResultsLoading && !searchResultsError) {
+      timerRef.current = setTimeout(() => {
+        setStillGeneratingDisplay(true);
+      }, '5000');
+    }
+    if (!searchResultsLoading) {
+      setStillGeneratingDisplay(false);
+      clearTimeout(timerRef.current);
+    }
+    if (searchResultsError) {
+      setStillGeneratingDisplay(false);
+      clearTimeout(timerRef.current);
+      setLoading(false);
+    }
+  }, [searchResultsError, searchResultsLoading, setLoading]);
```

Comment:
The message about loading taking a long time ideally should only have to depend on loading. Did loading continue for a noticeable time after an error message appeared?

Reply:
Yes, totally... the error would appear, and loading would just continue to stay true.

Reply:
Fixing the bug in useItisSearch should mean that you can simplify this to not depend on `searchResultsError` or `setLoading`